lib/data/repositories/pokemon_repository.dart

```diff
@@ -16,6 +18,27 @@ class PokemonDefaultRepository extends PokemonRepository {
   final GithubDataSource githubDataSource;
   final LocalDataSource localDataSource;
 
+  @override
+  Future<List<Pokemon>> getAllPokemons() async {
+    final hasCachedData = await localDataSource.hasData();
+
+    if (!hasCachedData) {
+      final pokemonGithubModels = await githubDataSource.getPokemons();
+      final pokemonHiveModels = pokemonGithubModels.map((e) => e.toHiveModel());
+
+      await localDataSource.savePokemons(pokemonHiveModels);
+    }
```

Comment:
@Matttx Since this part of your other PR has been split, which PR do you want to be merged first?

Reply:
This one because I need the getAllPokemonsUseCase for the other one 😄

Reply:
@Matttx Oh, okay!